game/src/main/kotlin/world/gregs/voidps/world/activity/dnd/shootingstar/ShootingStar.kts

```diff
@@ -1,186 +1,146 @@
     return false
 }
 
-fun startCrashedStarEvent() {
-    val starLocation = getRandomLocation()
-    val starToSpawn = getInitialStar()
-    currentStarTile = starLocation.location
-    currentActiveObject = GameObject(starToSpawn, currentStarTile, 10, 0)
-    objects.add(currentActiveObject)
-    println("${starLocation.locationDescription}")
-}
-
 fun handleMinedStarDust(currentMinedStar: GameObject) {
     val starPayout = currentMinedStar.def["collect_for_next_layer", -1]
-    if(totalCollected >= starPayout){
+    if (totalCollected >= starPayout) {
         val stage = currentMinedStar.id.takeLast(1)
-        if(stage.equals("1")){ //if is the lowest tier star just delete the star and clean the event
-            currentMinedStar.remove(currentMinedStar.intId, true)
+        if (stage.equals("1")) {
+            currentMinedStar.remove()
             cleanseEvent(false)
             return
         }
         val nextStage = currentMinedStar.id.replace(stage, (stage.toInt() - 1).toString())
-        val nextstar = currentMinedStar.replace(nextStage)
+        val nextStar = currentMinedStar.replace(nextStage)
         totalCollected = 0
-        changeStar(currentMinedStar.id, nextstar.id)
+        changeStar(currentMinedStar.id, nextStar.id)
+    }
+}
+
+fun checkIfEarlyBird(player: Player): Boolean {
+    if(earlyBird == null){
+        earlyBird = player
+        return true
+    }
+    return false
+}
+
+fun addStarDustCollected(){
+    totalCollected ++
+}
+
+fun startCrashedStarEvent() {
+    currentStarTile = StarLocationData.entries.random().location
+    val shootingStarShadow: NPC? = npcs.add("shooting_star_shadow", Tile(currentStarTile.x, currentStarTile.y + 6), Direction.NONE)
+    shootingStarShadow?.walkTo(currentStarTile, true, true)
+    World.queue("awaiting_shadow_walk", 6) {
+        val shootingStarObjectFalling: GameObject = objects.add("crashed_star_falling_object", currentStarTile)
+        World.queue("falling_star_object_removal", 1) {
+            players.forEach { player -> // theres probably a way to iterate through players in a region by id
+                if(player.tile.region == currentStarTile.region){ // make sure that the player is in the same region as the rock
+                    if(player.tile.equals(currentStarTile)){ // if the player tile is the same as rock tile
+                        val actual = currentStarTile
+                        val direction = player.tile.delta(actual.add(1,1)).toDirection()
+                        val delta = direction.delta
+                        player.damage(random.nextInt(10, 50))
+                        if (!player.blocked(direction)) {
+                            player.walkTo(Tile(currentStarTile.x + delta.x, currentStarTile.y + delta.y), true) // had to set 'noCollision' to true otherwise the star object itself was blocking the walk and forceWalk didn't look good.
+                        }
+                    }
+                }
+            }
```

Comment:
```suggestion
            for (tile in currentStarTile.toCuboid(2, 2)) {
                for (player in players[tile]) {
                    player.damage(random.nextInt(10, 50))
                    val direction = if (player.tile == object.tile) Direction.SOUTH else currentStarTile.delta(player.tile).toDirection()
                    if (!target.blocked(direction)) {
                        player.forceWalk(direction.delta, 0, direction.inverse()) // TODO check delay
                    }
                }
            }
```

Force walk/movement should look more smooth than walking, in osrs there's an animation but I've been unable to find it in rs2 (if it even exists). Perhaps 11945 or 4172 will be a decent replacement?

Reply:
i used 4172

![starfallplayeranim](https://github.com/GregHib/void/assets/79747812/7c167c84-da43-448b-ad16-93f8a3160805)